Here is the page that has list of all available bert models on tensorflow hub that one can download and make use of.

https://tfhub.dev/google/collections/bert/1

Here is the information on basic uncased BERT model,

https://tfhub.dev/tensorflow/bert_en_uncased_L-12_H-768_A-12/4

It uses L=12 hidden layers (i.e., Transformer blocks), a hidden size of H=768, and A=12 attention heads. This model has been pre-trained for English on the Wikipedia and BooksCorpus. 

In [3]:
import tensorflow_hub as hub
import tensorflow_text as text

In [11]:
preprocess_url = "https://tfhub.dev/tensorflow/bert_en_uncased_preprocess/3"
encoder_url = "https://tfhub.dev/tensorflow/bert_en_uncased_L-12_H-768_A-12/4"

In [12]:
bert_preprocess_model = hub.KerasLayer(preprocess_url)

In [13]:
text_test = ['nice movie indeed','I love python programming']
text_preprocessed = bert_preprocess_model(text_test)
text_preprocessed.keys()

dict_keys(['input_mask', 'input_type_ids', 'input_word_ids'])

In [14]:
text_preprocessed['input_mask']

<tf.Tensor: shape=(2, 128), dtype=int32, numpy=
array([[1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
       [1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]],
      dtype=int32)>

In [15]:
text_preprocessed['input_type_ids']

<tf.Tensor: shape=(2, 128), dtype=int32, numpy=
array([[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
       [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]],
      dtype=int32)>

In [16]:
text_preprocessed['input_word_ids']

<tf.Tensor: shape=(2, 128), dtype=int32, numpy=
array([[  101,  3835,  3185,  5262,   102,     0,     0,     0,     0,
            0,     0,     0,     0,     0,     0,     0,     0,     0,
            0,     0,     0,     0,     0,     0,     0,     0,     0,
            0,     0,     0,     0,     0,     0,     0,     0,     0,
            0,     0,     0,     0,     0,     0,     0,     0,     0,
            0,     0,     0,     0,     0,     0,     0,     0,     0,
            0,     0,     0,     0,     0,     0,     0,     0,     0,
            0,     0,     0,     0,     0,     0,     0,     0,     0,
            0,     0,     0,     0,     0,     0,     0,     0,     0,
            0,     0,     0,     0,     0,     0,     0,     0,     0,
            0,     0,     0,     0,     0,     0,     0,     0,     0,
            0,     0,     0,     0,     0,     0,     0,     0,     0,
            0,     0,     0,     0,     0,     0,     0,     0,     0,
            0,     0,     0, 

In [17]:
bert_model = hub.KerasLayer(encoder_url)

In [18]:
bert_results = bert_model(text_preprocessed)

In [19]:
bert_results.keys()

dict_keys(['pooled_output', 'default', 'sequence_output', 'encoder_outputs'])

In [20]:
bert_results['sequence_output']

<tf.Tensor: shape=(2, 128, 768), dtype=float32, numpy=
array([[[ 0.07292041,  0.08567824,  0.14476828, ..., -0.09677109,
          0.08722142,  0.07711099],
        [ 0.17839399, -0.1900608 ,  0.50349444, ..., -0.05869823,
          0.3271716 , -0.15578537],
        [ 0.18701471, -0.4338876 , -0.4887512 , ..., -0.15502733,
          0.00145092, -0.24470995],
        ...,
        [ 0.12083066,  0.1288423 ,  0.46453524, ...,  0.07375526,
          0.17441979,  0.16522065],
        [ 0.07967877, -0.011907  ,  0.5022543 , ...,  0.13777767,
          0.21002221,  0.00624587],
        [-0.07212684, -0.2830348 ,  0.5903337 , ...,  0.47551972,
          0.16668478, -0.08920318]],

       [[-0.07900587,  0.36335114, -0.21101567, ..., -0.17183784,
          0.16299735,  0.6724267 ],
        [ 0.27883482,  0.43716273, -0.357647  , ..., -0.04463646,
          0.3831518 ,  0.58879876],
        [ 1.2037675 ,  1.0727016 ,  0.4840882 , ...,  0.24921009,
          0.40730932,  0.40481773],
        ...,

In [21]:
bert_results['pooled_output']

<tf.Tensor: shape=(2, 768), dtype=float32, numpy=
array([[-0.7917742 , -0.21411929,  0.49769515, ...,  0.24465188,
        -0.47334474,  0.81758714],
       [-0.9171231 , -0.4793519 , -0.7865699 , ..., -0.61751735,
        -0.71026874,  0.92184305]], dtype=float32)>

**101 --> CLS token**

**102 --> SEP token**

BERT uses CLS as a special token at the begining of each setence whereas SEP as a special token to 
separate two sentences or end sinle sentece

In [22]:
len(bert_results['encoder_outputs'])

12

Since we are using BERT base model it has 12 encoder layers, that's why above the length of encoder_output is 12.
Read more about purpose of each element here: https://tfhub.dev/tensorflow/bert_en_uncased_L-12_H-768_A-12/4

You can see below that last element of encoder_outputs is basically a sequence_output

In [23]:
bert_results['encoder_outputs'][-1] == bert_results['sequence_output']

<tf.Tensor: shape=(2, 128, 768), dtype=bool, numpy=
array([[[ True,  True,  True, ...,  True,  True,  True],
        [ True,  True,  True, ...,  True,  True,  True],
        [ True,  True,  True, ...,  True,  True,  True],
        ...,
        [ True,  True,  True, ...,  True,  True,  True],
        [ True,  True,  True, ...,  True,  True,  True],
        [ True,  True,  True, ...,  True,  True,  True]],

       [[ True,  True,  True, ...,  True,  True,  True],
        [ True,  True,  True, ...,  True,  True,  True],
        [ True,  True,  True, ...,  True,  True,  True],
        ...,
        [ True,  True,  True, ...,  True,  True,  True],
        [ True,  True,  True, ...,  True,  True,  True],
        [ True,  True,  True, ...,  True,  True,  True]]])>

In [24]:
bert_results['encoder_outputs']

[<tf.Tensor: shape=(2, 128, 768), dtype=float32, numpy=
 array([[[ 0.12901431,  0.0064473 , -0.03614966, ...,  0.04999612,
           0.06149197, -0.0265754 ],
         [ 1.1753383 ,  1.2140788 ,  1.1569982 , ...,  0.1163439 ,
          -0.35855347, -0.40490162],
         [ 0.03859065,  0.5386996 , -0.21089783, ...,  0.21858186,
           0.72601694, -1.1158606 ],
         ...,
         [-0.07587012, -0.2542191 ,  0.7075511 , ...,  0.5054202 ,
          -0.18878667,  0.15028363],
         [-0.16066603, -0.2808969 ,  0.57597065, ...,  0.5275854 ,
          -0.11141386,  0.0288756 ],
         [-0.04428149, -0.20279574,  0.59093535, ...,  0.8133834 ,
          -0.39075795, -0.02601741]],
 
        [[ 0.18903592,  0.02752541, -0.06513745, ..., -0.00620207,
           0.15053886,  0.03165449],
         [ 0.5916149 ,  0.75891405, -0.07240671, ...,  0.6190398 ,
           0.8292888 ,  0.16161953],
         [ 1.4460826 ,  0.44602653,  0.40990287, ...,  0.4825589 ,
           0.62691164,  0.13